checkout/signals.py

```diff
@@ -0,0 +1,21 @@
+from django.db.models.signals import post_save, post_delete
+from django.dispatch import receiver
+
+from .models import OrderLineItem
+
+
+@receiver(post_save, sender=OrderLineItem)
+def update_on_save(sender, instance, created, **kwargs):
+	"""
+	Update order total on update of individual lineitem,
+	e.g. initial creation, updating quantity, etc
+	"""
+	instance.order.update_total()
+
+
+@receiver(post_delete, sender=OrderLineItem)
```

Comment:
Seeing that it's the same exact functionality, wouldn't it be better to attach both decorators to this function?

Reply:
Hmmm, possibly. Only issue is `post_save` sends the `created` argument, so would that puke on `post_delete` if it's not sent? I will test it out. I felt separating them just makes it more clear for the students exactly what's going on even though you're right, they're the same functionality.